zha/application/platforms/select.py

```diff
@@ -820,3 +820,55 @@ class DanfossViewingDirection(ZCLEnumSelectEntity):
     _attribute_name = "viewing_direction"
     _attr_translation_key: str = "viewing_direction"
     _enum = danfoss_thermostat.DanfossViewingDirectionEnum
+
+
+class SinopeLightLedColors(types.enum32):
+    """Color values for Sinope light switch status LEDs."""
+
+    Lim = 0x0AFFDC
+    Amber = 0x000A4B
+    Fushia = 0x0100A5
+    Perle = 0x64FFFF
+    Blue = 0xFFFF00
+
+
+@CONFIG_DIAGNOSTIC_MATCH(
+    cluster_handler_names="sinope_manufacturer_specific",
+    models={
+        "DM2500ZB",
+        "DM2500ZB-G2",
+        "DM2550ZB",
+        "DM2550ZB-G2",
+        "SW2500ZB",
+        "SW2500ZB-G2",
+    },
+)
+class SinopeLightLEDOffColorSelect(ZCLEnumSelectEntity):
+    """Representation of the marker LED color off-state color of Sinope light switches."""
+
+    _unique_id_suffix = "off_led_color"
+    _attribute_name = "off_led_color"
+    _attr_translation_key: str = "off_led_color"
+    _enum = SinopeLightLedColors
+    _attr_icon: str = "mdi:palette-outline"
+
+
+@CONFIG_DIAGNOSTIC_MATCH(
+    cluster_handler_names="sinope_manufacturer_specific",
+    models={
+        "DM2500ZB",
+        "DM2500ZB-G2",
+        "DM2550ZB",
+        "DM2550ZB-G2",
+        "SW2500ZB",
+        "SW2500ZB-G2",
+    },
+)
+class SinopeLightLEDOnColorSelect(ZCLEnumSelectEntity):
+    """Representation of the marker LED color on-state color of Sinope light switches."""
+
+    _unique_id_suffix = "on_led_color"
+    _attribute_name = "on_led_color"
+    _attr_translation_key: str = "on_led_color"
+    _enum = SinopeLightLedColors
+    _attr_icon: str = "mdi:palette"
```

Comment:
@dmulcahey 

I can't seem to get HA to respect the `_attr_icon` attribute in the UI. It's showing the default entity icon instead. 
Any idea what I'm missing here? 

<img width="322" alt="image" src="https://github.com/user-attachments/assets/730c8a1a-3c0a-4d1e-8894-c794f84e51c9">

Reply:
Looks like a bug: <https://github.com/zigpy/zha/blob/f015eab0514dc33972d97535958fc78aeafce853/zha/application/platforms/__init__.py#L139>

Reply:
https://github.com/zigpy/zha/pull/171

Reply:
Thanks for the quick patch!

Reply:
Core has moved to `icons.json` to specify entity icons and ZHA has exactly one spot left that uses them. You should be able to add your entity's icon to the same file: <https://github.com/home-assistant/core/blob/dev/homeassistant/components/zha/icons.json>. If you update the PR with a diff, we can include it when the ZHA library is bumped.

Reply:
Something like this? 

```diff
diff --git a/homeassistant/components/zha/icons.json b/homeassistant/components/zha/icons.json
index 9b060e8105..bb4224f2bd 100644
--- a/homeassistant/components/zha/icons.json
+++ b/homeassistant/components/zha/icons.json
@@ -94,6 +94,12 @@
       },
       "keypad_lockout": {
         "default": "mdi:lock"
+      },
+      "off_led_color": {
+        "default": "mdi:palette-outline"
+      },
+      "on_led_color": {
+        "default": "mdi:palette"
       }
     },
     "sensor": {
@@ -158,6 +164,9 @@
       },
       "hooks_locked": {
         "default": "mdi:lock"
+      },
+      "double_up_full": {
+        "default": "mdi:gesture-double-tap"
       }
     }
   },
```

I'm assuming I should remove the `_attr_icon` from the relevant entities.

Reply:
Tested the change locally with the diff above, works great!

<img width="312" alt="image" src="https://github.com/user-attachments/assets/5ea217d5-cefd-497b-bfbe-e00588df20dc">